ref_builder/repo.py

```diff
@@ -249,22 +249,29 @@ def use_transaction(self) -> Generator[Transaction, None, None]:
 
         try:
             yield self._transaction
+
         except Exception as e:
             self.prune()
 
-            if not isinstance(e, AbortTransactionError):
-                raise
-        else:
-            self._transaction = None
+            logger.error("Error encountered during transaction.", error=e)
```

Comment:
When the exception is reraised, `finally` will still be executed. Logging versus reraising is not going to change that.

Logging instead of reraising means the exception cannot be caught outside the transaction context and we are going to ignore all exceptions that occur in transactions and continue execution. The default behaviour should be that an exception should stop execution and rollback any events ahead of head. Specific, expected exceptions can be suppressed outside the transaction context.

These changes here nerf transaction abortion. This is not necessarily an error and should not be logged as such. That is why there is a type check for it in the `except` block.